### ISSUE TITLE
Vulnerability in RestSharp

### ISSUE THREAD

**Issue by RobJDavey** [eBay/ebay-oauth-csharp-client#46]
The current release of `eBay.OAuth.Client` has a dependency on RestSharp >= 110.2.0.
RestSharp versions < 112.0.0 have a vulnerability in them. See: https://github.com/advisories/GHSA-4rr6-2v9v-wcpc

This results in a build time warning:
```
warning NU1902: Package 'RestSharp' 110.2.0 has a known moderate severity vulnerability, https://github.com/advisories/GHSA-4rr6-2v9v-wcpc
```

While the >= 110.2.0 doesn't prevent you from upgrading to 112.0, upgrading to this version results in

```
Method not found: 'RestSharp.RestResponse RestSharp.RestClientExtensions.Execute(RestSharp.RestClient, RestSharp.RestRequest, System.Threading.CancellationToken)'.
```

due to breaking changes within RestSharp.

As a result, it's currently not possible for anyone using the `eBay.OAuth.Client` to remove this warning from their codebase.

Due to how often RestSharp updates result in breaking changes like this between major versions, ideally the dependency on RestSharp should be removed entirely, but for now an update to the version of RestSharp to the one without the vulnerability will at least remove this warning.

**Comment by UkeHa** [comment]
@RobJDavey, this should be fixed with #33

**Comment by RobJDavey** [comment]
@UkeHa I'm not sure what you mean? The issue I'm reporting here is a new vulnerability that affects RestSharp 110.2.0, which is the version used by the current release of `eBay.OAuth.Client`.

**Comment by UkeHa** [comment]
@RobJDavey, yes and my patch updates it to 112

**Comment by RobJDavey** [comment]
@UkeHa oh I see… I think you meant to say #44 😂

**Comment by LokeshRishi** [comment]
Fixed in version [2.0.3](https://github.com/eBay/ebay-oauth-csharp-client/pull/48)
